airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -211,7 +212,7 @@ def get_dag_runs_batch(session):
         raise BadRequest(detail=str(err.messages))
 
     appbuilder = current_app.appbuilder
-    readable_dag_ids = appbuilder.sm.get_readable_dag_ids(g.user)
+    readable_dag_ids = {dag.dag_id for dag in appbuilder.sm.get_readable_dags(g.user)}
     query = session.query(DagRun)
     if data.get("dag_ids"):
         dag_ids = set(data["dag_ids"]) & set(readable_dag_ids)
```

Comment:
```suggestion
        dag_ids = set(data["dag_ids"]) & readable_dag_ids
```

Because `readable_dag_ids` is now already a set

Or

```suggestion
        dag_ids = readable_dag_ids.intersection(data["dag_ids"])
```